services/spar/test-integration/Util/SCIM.hs

```diff
@@ -68,16 +78,33 @@ registerSCIMToken teamid midpid = do
           }
   pure tok
 
--- | Generate a SCIM user with a random name and handle.
-randomSCIMUser :: TestSpar SCIM.User.User
+-- | Generate a SCIM user with a random name and handle.  At the very least, everything considered
+-- in @instance UserShouldMatch SCIM.User.User User@ and @validateSCIMUser@ must be random here.
+-- FUTUREWORK: make this more exhaustive.  change everything that can be changed!  move this to the
+-- hspec package when done.
+randomSCIMUser :: MonadRandom m => m SCIM.User.User
 randomSCIMUser = do
-    suffix <- pack <$> replicateM 5 (liftIO (randomRIO ('0', '9')))
+    suffix <- cs <$> replicateM 5 (getRandomR ('0', '9'))
+    emails <- replicateM 3 randomSCIMEmail
     pure $ SCIM.User.empty
         { SCIM.User.userName    = "scimuser_" <> suffix
         , SCIM.User.displayName = Just ("Scim User #" <> suffix)
         , SCIM.User.externalId  = Just ("scimuser_extid_" <> suffix)
+        , SCIM.User.emails      = Just emails
         }
 
+randomSCIMEmail :: MonadRandom m => m Email.Email
+randomSCIMEmail = do
+    let typ     :: Maybe Text = Nothing
+        primary :: Maybe Bool = Nothing  -- TODO: where should we catch users with more than one
+                                         -- primary email?
+    value :: Email.EmailAddress2 <- do
+      localpart  <- cs <$> replicateM 15 (getRandomR ('a', 'z'))
```

Comment:
(I know my code wasn't testing them. In hindsight this seems like a lazy decision.)

Reply:
i think it's fine.  i vaguely remember having seen we're using `+` for making tests work somewhere, but everything but the receiving MTA should not handle them any different from non-plus-carrying emails.

Reply:
ok, i was wrong: wire needs to know whether two emails go the same recipient, which requires stripping the `+` part.  i still don't think scim should worry about this, because (at least in my model of what we're doing) we neither validate nor use the emails we store.  this question will become relevant as soon as we do, though.